src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<T> {
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<T> getAll(Class<? extends T> type) {
+        List<T> machines = new ArrayList<>();
+
+        if (Truck.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new TruckProducer().get());
+        } else if (Bulldozer.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new BulldozerProducer().get());
+        } else if (Excavator.class.isAssignableFrom(type)) {
+            machines.addAll((List<T>) new ExcavatorProducer().get());
+        }
+
+        return machines;
```

Comment:
Consider creating a local variable in your implementation of `getAll(Class type)` method to hold the result from the producer's `get()` method before adding it to the `machines` list.